### PR TITLE
Add artifactory & SDK EE artifacts to split-release process

### PR DESCRIPTION
**Pull request by cocreature**
Apologies for doing two things in one PR but they seemed somewhat
entangled.

For artifactory we follow the approach we use for Maven artifacts and
publish there directly.

For the SDK EE tarball, we just throw it in the split-release
directory and then the assembly repo can pick it up.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/publish-artifactory.sh [hunk 1]
```diff
@@ -51,11 +51,14 @@ for base in non-repudiation-core non-repudiation-client; do
...
-for platform in linux macos windows; do
-    EE_TARBALL=daml-sdk-$RELEASE_TAG-$platform-ee.tar.gz
-    push sdk-ee $EE_TARBALL
-    push sdk-ee $EE_TARBALL.asc
-done
-EE_INSTALLER=daml-sdk-$RELEASE_TAG-windows-ee.exe
-push sdk-ee $EE_INSTALLER
-push sdk-ee $EE_INSTALLER.asc
+# For the split release process these are not published to artifactory.
+if [[ "$#" -lt 3 || $3 != "split" ]]; then
```
Comment (cocreature) [inline ci/publish-artifactory.sh:55]: this is a bit ugly but given that I’m hoping we can just drop the non-split release process in a few weeks and just delete this, this seems better than duplicating the whole file.

Comment (garyverhaegen-da) [reply ci/publish-artifactory.sh:55]: Since YAML files are from main, will we still be able to do patch releases if we drop the non-split process?

Comment (cocreature) [reply ci/publish-artifactory.sh:55]: I’m not following, this is not a yaml file it will never be used for old releases.

Comment (cocreature) [reply ci/publish-artifactory.sh:55]: oh sorry, I think I was just suggesting we can drop it from this file not from the yaml files. Does that make more sense?

Comment (garyverhaegen-da) [reply ci/publish-artifactory.sh:55]: Yes, that does. 👍

**Review by garyverhaegen-da — APPROVED** [review]

**Pull request merged**